optuna_dashboard/ts/components/TrialList.tsx

```diff
@@ -442,6 +468,7 @@ export const TrialList: FC<{ studyDetail: StudyDetail | null }> = ({
     estimateSize: () => 73.31,
     overscan: 10,
   })
+  const [filterInput, setFilterInput] = useState(trialFilterQuery)
```

Comment:
Currently, every time TextField updates, it triggers re-render of the React component. This results in performance issue.  Could you use `useDefferedValue()` hook for the debouncing?
![review-optuna-dashboard](https://github.com/user-attachments/assets/caf0d545-7916-4c34-afd9-834f316dbdec)

Usage is like below:

https://github.com/optuna/optuna-dashboard/blob/79e431fce57322b1ad13ad0f4b8cfa97e73c878c/optuna_dashboard/ts/components/StudyList.tsx#L53-L54

Please refer to following pages for details about the debouncing.

* https://developer.mozilla.org/en-US/docs/Glossary/Debounce
* https://react.dev/reference/react/useDeferredValue

Reply:
Thank you for pointing it out. I'm not aware of it.
I applied `useDeferredValue` to `trialFilterQuery` in the commit https://github.com/optuna/optuna-dashboard/pull/1127/commits/a132e22b74fa43a06feed21f9964050115bf3f6d.